### ISSUE TITLE
Investigate memory footprint of internal mz views including mz_metrics* 

### ISSUE THREAD

**Issue by krishmanoh2** [MaterializeInc/materialize#7438]
Customers have reported significant memory usage reduction when setting introspection-frequency=off or even 60seconds. The default is 1second.

![Screenshot from 2021-07-15 12-58-03](https://user-images.githubusercontent.com/72518904/125835185-b2ae8d0f-9831-4e44-a560-0eed8a679c50.png)
![Screenshot from 2021-07-15 12-57-31](https://user-images.githubusercontent.com/72518904/125835205-953adb47-b666-44bc-b284-e5344fbaa359.png)

Can we identify the cause behind the higher memory usage?

**Comment by krishmanoh2** [comment]
We maybe able to repro this using the avro upsert test. Creating a few dependent materializations and corresponding tail should be close to actual workload.

**Comment by uce** [comment]
cc @antiguru @antifuchs If I remember correctly, you were onboard with increasing the mz_metrics-related `introspection-frequency`, right?

**Comment by antifuchs** [comment]
Yes, absolutely on board - any interval will do, and I think something on the order of 20s-1min should cut down nicely on the amount of memory this eats.
